project/apps/web/lib/api/auth.ts

```diff
@@ -10,12 +12,42 @@ export interface UserDetails {
   username: string;
 }
 
-export const signUp = (signUpDto: SignUpDto) =>
-  apiCall("post", "/auth/signup", signUpDto);
-
-export const signIn = (signInDto: SignInDto) =>
-  apiCall("post", "/auth/signin", signInDto);
-
-export const signOut = () => apiCall<void>("post", "/auth/signout");
+interface AuthState {
+  user: UserDetails | null;
+  signUp: (signUpDto: SignUpDto) => Promise<void>;
+  signIn: (signInDto: SignInDto) => Promise<void>;
+  signOut: () => Promise<void>;
+  fetchUser: () => Promise<void>;
+  refreshAccessToken: () => Promise<void>;
+}
 
-export const me = (): Promise<UserDetails> => apiCall("get", "/auth/me");
+export const useAuthStore = create<AuthState>()(
```

Comment:
should this live within a `hooks` or `store` folder instead? then we can import this api calls